### DISCUSSION TITLE
EFX rule conflict with BT-610-Procedure-Buyer and BT-10-Procedure Buyer on Notice Type 2

### DISCUSSION THREAD

**Discussion by lucaCorigliano-SA** [OP-TED/eForms-SDK#729] [Q&A] [answered]
The `GR-Buyer` sections contains two fields: `BT-610-Procedure-Buyer` and `BT-10-Procedure-Buyer`, which are both **mandatory** and both produce a `cbc:ActivityTypeCode` XML element in the Physical Model.

Normally, they have an EFX rule that will make them **forbidden** them based on which option is selected in the `BT-11-Procedure-Buyer` select, which solves a situation in which we have two distinct `cbc:ActivityTypeCode` elements that would create a validation issue.

We noticed, though, that in Notice Type 2, the **forbidden** rule for `BT-610-Procedure-Buyer` doesn't apply as you can see below:
```"noticeTypes" : [ "3", "6", "9", "13", "14", "18", "19", "21", "26", "27", "28", "31", "32", "34", "35" ],```

This will make it so that when you select any option in `BT-11-Procedure-Buyer` from the following list: `'pub-undert','pub-undert-cga','pub-undert-la','pub-undert-ra','spec-rights-entity'`, the Form will **never be able to be validated** correctly since both BT-610 and BT-10 will be **not forbidden** and **mandatory**. 

**This issue is present in both SDKs 1.8.0 and 1.9.0.**

For now our workaround has been to apply a custom modification in `fields.json` adding noticeType 2 in the forbidden rule for `BT-610-ProcedureBuyer`.

**Comment by YvesJo** [comment, marked as answer]
Hi Luca,
Thanks for getting in touch with your concerns.
There is however no issue with the rules in place as BT-10 and BT-610 may coexist. D25 is normally for contracting entities only, however in some cases, contracting authorities may act as contracting entities, in which case they have to specify their activity as contracting authority in addition to their activity while acting as a contracting entity. This is not straightforward and is however well summarized in the Regulation Annex where BT-610 appears as purely Mandatory and BT-10 as conditionally mandatory depending on the Buyer Legal Type.
In field.json for BT-10-Procedure:

>     "forbidden" : {
>       "value" : false,
>       "severity" : "ERROR",
>       "constraints" : [ {
>         "noticeTypes" : [ "22", "38", "39", "40", "X01", "X02" ],
>         "value" : true,
>         "severity" : "ERROR"
>       }, {
>         "noticeTypes" : [ "2", "3", "5", "6", "8", "9", "11", "12", "13", "14", "15", "17", "18", "19", "20", "21", "24", "25", "26", "27", "28", "30", "31", "32", "33", "34", "35", "37", "T01", "T02" ],
>         "condition" : "{ND-ContractingParty} ${BT-11-Procedure-Buyer not in ('body-pl','body-pl-cga','body-pl-la','body-pl-ra','cga','def-cont','eu-ins-bod-ag','grp-p-aut','int-org','la','org-sub','org-sub-cga','org-sub-la','org-sub-ra','ra')}",
>         "value" : true,
>         "severity" : "ERROR"
>       } ]
>     },
>     "mandatory" : {
>       "value" : false,
>       "severity" : "ERROR",
>       "constraints" : [ {
>         "noticeTypes" : [ "1", "2", "4", "5", "7", "8", "10", "11", "14", "15", "16", "17", "19", "23", "24", "29", "30", "32", "35", "36", "37", "CEI" ],
>         "value" : true,
>         "severity" : "ERROR"
>       }

the mandatory rules not having an associated condition and for which a conditionally forbidden rule exists (this is the case for form 2), share the same condition as for the forbidden rule, however in its negated expression.
For form 2:

- BT-10-Procedure is forbidden if "condition",
- BT-10-Procedure is mandatory if "not(condition)"
This because, to be mandatory, a field must first be allowed (i.e. not forbidden)

BT-610-Procedure for form 2 is just Mandatory without any condition.
I hope this helps better understand.
